include/engine/api/route_api.hpp

```diff
@@ -149,6 +149,7 @@ class RouteAPI : public BaseAPI
                                                               leg_geometry,
                                                               phantoms.source_phantom,
                                                               phantoms.target_phantom);
+                leg.steps = guidance::anticipateLaneChange(std::move(leg.steps));
                 leg_geometry = guidance::resyncGeometry(std::move(leg_geometry), leg.steps);
```

Comment:
For roundabout handling, we might either have to handle the lanes within the roundabout post-processing already or prior to it, since otherwise intermediate instructions are lost.
Is your logic in a state that would allow handling of arbitrary instruction streams having no-lane information in between?